addons/mrp/mrp.py

```diff
@@ -167,6 +173,11 @@ class mrp_bom(osv.osv):
             domain="['&', ('product_tmpl_id','=',product_tmpl_id), ('type','!=', 'service')]",
             help="If a product variant is defined the BOM is available only for this product."),
         'bom_line_ids': fields.one2many('mrp.bom.line', 'bom_id', 'BoM Lines', copy=True),
+        'categ_id': fields.related('product_tmpl_id', 'categ_id', type='many2one', relation='product.category', string='Product Category', readonly=True,
+            store = {
+                'mrp.bom': (lambda self, cr, uid, ids, c={}: ids, ['product_tmpl_id'], 10),
```

Comment:
c={} is incorrect, but has no impact here. Always do c=None in function signature